src/store/movie/saga.js

```diff
@@ -1,15 +1,20 @@
 import { put, call, takeLatest, take } from "redux-saga/effects";
 import { movieService } from "../../services/MovieService";
 import {
-    addMovie,
-    getMovies,
-    setMovies
+  addMovie,
+  getMovies,
+  setMovies,
+  setPaginated
 } from "./slice";
- 
-function* getMoviesHandler() {
+
+function* getMoviesHandler(action) {
   try {
-    const movies = yield call(movieService.getAll);
+    const movies = yield call(movieService.getAll, action.payload.page);
+    if (action.payload?.page > 1) {
+      yield put(setPaginated(movies));
```

Comment:
You can use the same setter (setMovies) for the first page and other pages, check comment below